src/components/newGrid.jsx

```diff
@@ -27,6 +39,7 @@ export default class NewGrid extends React.Component {
   }
 
   handleClick(event){
+    this.state.showModal = false;
```

Comment:
I would swap the order of this and the line below just for logical order -- 
if grid creation doesn't work for whatever reason, the modal should just be left open so the user knows it didn't work.